Fix dynamo func defaults handling for torch.device, size, dtype

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92880

Previously, these torch types were not handled in the wrap_bound_arg
handler.

Add a unit test and verify it is fixed.

Fixes #91084

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

wconstab committed:
Fix dynamo func defaults handling for torch.device, size, dtype

Previously, these torch types were not handled in the wrap_bound_arg
handler.

Add a unit test and verify it is fixed.

Fixes #91084

[ghstack-poisoned]

ezyang reviewed (approved):
But this seems suspicious. Why do we have to allowlist the types one by one?

wconstab commented:
> But this seems suspicious. Why do we have to allowlist the types one by one?

yea, I am wondering if we should be cutting the whole 'wrap_bound_arg' thing out and replacing it with a call into other infra like VariableBuilder.  I'm not sure if/why we want it to stay separate, but it probably isn't going to work if we just cut over.  Anybody know the history or have a design opinion?  @voznesenskym  @jansel

wconstab committed:
Update on "Fix dynamo func defaults handling for torch.device, size, dtype"


Previously, these torch types were not handled in the wrap_bound_arg
handler.

Add a unit test and verify it is fixed.

Fixes #91084

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge
